### PR TITLE
Type select pokedex mobile

### PR DESCRIPTION
**Pull request by DeWolfRobin**
I created a working piece of code, I recommend some refactoring but since I'm new to these languages, this is the cleanest I got it.

### CODE REVIEW

**Comment by DeWolfRobin** [comment]
there appears to be 1 minor bug, whenever you go beyond the first page, the application crashes upon click.

File: less/style.less [hunk 1]
```diff
@@ -8,6 +8,18 @@ html {
...
+  top: 180px;
+  right: 125px;
+}
+
+.OffensePreview {
+  position: fixed;
+  top: 180px;
+  left: 125px;
+}
+
```
Comment (wavebeem) [inline less/style.less]: This CSS is no longer used.

**Comment by DeWolfRobin** [comment]
I now realize it is probably because of the Type.NONE in the matchups

File: src/Dex.tsx [hunk 2]
```diff
@@ -40,8 +50,13 @@ function makePKMN(p: Pokemon, i: number) {
   );
   const displayNumber = "#" + String(p.number).padStart(3, "0");
   const style = { minHeight: "100px" };
+  const onClick = () => {
+    gChangeTab(1);
+    gupdateType1(p.types[0]);
+    gupdateType2(p.types[1]);
```
Comment (wavebeem) [inline src/Dex.tsx]: This right now is crashing if the Pokemon doesn't have a second type, you should use "none" as a fallback value.

**Comment by DeWolfRobin** [comment]
fixed the issue

File: src/Dex.tsx [hunk 3]
```diff
@@ -9,6 +9,11 @@ import { Pokemon } from "./pkmn";
 
 const PAGE_SIZE = 50;
 
+var gChangeTab: (index: number) => void;
+var gupdateType0: (index: Type) => void;
+var gupdateType1: (index: Type) => void;
+var gupdateType2: (index: Type) => void;
+
```
Comment (wavebeem) [inline src/Dex.tsx]: Not really what's up with the `g` prefix in all this stuff here

File: src/App.tsx [hunk 4]
```diff
@@ -9,7 +9,7 @@ import { AllPokemon } from "./pkmn";
 import Footer from "./Footer";
 
 function App() {
-  const [tab, changeTab] = React.useState(1);
+  var [tab, changeTab] = React.useState(2);
```
Comment (wavebeem) [inline src/App.tsx]: please keep this set to const :)

File: src/Dex.tsx [hunk 5]
```diff
@@ -60,9 +79,13 @@ interface DexProps {
...
 
 function Dex(props: DexProps) {
-  const { pkmn, search, updateSearch, updateCurrentPage, currentPage } = props;
+  const { changeTab,updateType0, updateType1, updateType2, pkmn, search, updateSearch, updateCurrentPage, currentPage } = props;
   const searchInput = <Search search={search} updateSearch={updateSearch} />;
   const emptyState = <p className="silver f1 b tc m0">no pokémon found</p>;
+  gChangeTab = changeTab;
+  gupdateType0 = updateType0;
+  gupdateType1 = updateType1;
+  gupdateType2 = updateType2;
```
Comment (wavebeem) [inline src/Dex.tsx]: I don't really understand the point in this assignment?

**Comment by wavebeem** [comment]
you don't need to close the PR, it's ok i just had a few questions

**Pull request closed without merging**